pkg/inst-api-semconv/instrumenter/http/http_metrics.go

```diff
@@ -0,0 +1,191 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/inst-api-semconv/instrumenter/utils"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.30.0"
+	"log"
+	"sync"
+	"time"
+)
+
+const http_server_request_duration = "http.server.request.duration"
+
+const http_client_request_duration = "http.client.request.duration"
+
+type HttpServerMetric struct {
+	key                   attribute.Key
+	serverRequestDuration metric.Float64Histogram
+}
+
+type HttpClientMetric struct {
+	key                   attribute.Key
+	clientRequestDuration metric.Float64Histogram
+}
+
+var mu sync.Mutex
+
+var httpMetricsConv = map[attribute.Key]bool{
+	semconv.HTTPRequestMethodKey:      true,
+	semconv.URLSchemeKey:              true,
+	semconv.ErrorTypeKey:              true,
+	semconv.HTTPResponseStatusCodeKey: true,
+	semconv.HTTPRouteKey:              true,
+	semconv.NetworkProtocolNameKey:    true,
+	semconv.NetworkProtocolVersionKey: true,
+	semconv.ServerAddressKey:          true,
+	semconv.ServerPortKey:             true,
+}
+
+var globalMeter metric.Meter
+
+func InitHttpMetrics(m metric.Meter) {
+	mu.Lock()
+	defer mu.Unlock()
+	globalMeter = m
+}
+
+func HttpServerMetrics(key string) *HttpServerMetric {
+	mu.Lock()
+	defer mu.Unlock()
+	return &HttpServerMetric{key: attribute.Key(key)}
+}
+
+func HttpClientMetrics(key string) *HttpClientMetric {
+	mu.Lock()
+	defer mu.Unlock()
+	return &HttpClientMetric{key: attribute.Key(key)}
+}
+
+// for test only
+func newHttpServerMetric(key string, meter metric.Meter) (*HttpServerMetric, error) {
+	m := &HttpServerMetric{
+		key: attribute.Key(key),
+	}
+	d, err := newHttpServerRequestDurationMeasures(meter)
+	if err != nil {
+		return nil, err
+	}
+	m.serverRequestDuration = d
+	return m, nil
+}
+
+func newHttpServerRequestDurationMeasures(meter metric.Meter) (metric.Float64Histogram, error) {
```

Comment:
This function has some duplicate logic with newHttpClientRequestDurationMeasures, we can extract it to a common function.